packages/util/src/signature.ts

```diff
@@ -110,20 +133,20 @@ export const toCompactSig = function (v: bigint, r: Buffer, s: Buffer, chainId?:
  * it's a signed message (EIP-191 or EIP-712) adding `27` at the end. Remove if needed.
  */
 export const fromRpcSig = function (sig: string): ECDSASignature {
-  const buf: Buffer = toBuffer(sig)
+  const bytes: Uint8Array = toBytes(sig)
 
-  let r: Buffer
-  let s: Buffer
+  let r: Uint8Array
+  let s: Uint8Array
   let v: bigint
-  if (buf.length >= 65) {
-    r = buf.slice(0, 32)
-    s = buf.slice(32, 64)
-    v = bufferToBigInt(buf.slice(64))
-  } else if (buf.length === 64) {
+  if (bytes.length >= 65) {
+    r = bytes.slice(0, 32)
+    s = bytes.slice(32, 64)
+    v = bytesToBigInt(bytes.slice(64))
+  } else if (bytes.length === 64) {
     // Compact Signature Representation (https://eips.ethereum.org/EIPS/eip-2098)
-    r = buf.slice(0, 32)
-    s = buf.slice(32, 64)
-    v = BigInt(bufferToInt(buf.slice(32, 33)) >> 7)
+    r = bytes.slice(0, 32)
+    s = Uint8Array.from(bytes.slice(32, 64))
```

Comment:
Is the extra Uint8Array.from necessary?

Reply:
Investigated this a bit further, and the [Uint8Array.slice spec](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/slice) mention that the method returns a new array. I therefore removed the Uint8Array.from. Tests are passing.